shell/server/proxy.js

```diff
@@ -1711,11 +1743,22 @@ class Proxy {
         if (request.method === "HEAD") {
           content.body.stream.close();
           response.rejectResponseStream(new Error("HEAD request; content doesn't matter."));
+        } else if (response.socket.destroyed) {
+          // The client has already closed the stream. No additional "close" event will be
+          // emitted. Annoyingly, we cannot rely on `response.write()` throwing an exception
+          // in this case, so if we don't handle it specially we could end up with a
+          // `ResponseStream` that gets stuck waiting for a "drain" event that never arrives.
+          content.body.stream.close();
+          response.rejectResponseStream(new Error("client disconnected"));
         } else {
+          const idx = this.streamingResponseCounter.toString();
```

Comment:
Stringifying idx here probably just means that V8 parses it again when you use it as an index below. Integer indexes (including stringified integers) are handled specially by V8.